### ISSUE TITLE
View Candidate Attachments in TC

### ISSUE THREAD

**Issue by sadatmalik** [talentbeyondboundaries/tbbtalentv2#80]
As a User 
I would like Candidate Attachments to be visible in TC 
So they are not downloaded to my device

**Comment by cazcam34** [comment]
Open in browser not download

**Comment by cazcam34** [comment]
I believe this has been addressed with the new preview CV tab in the candidate side panel. Unless this is also to be rolled out to all candidate attachments under the additional info tab?

**Comment by sadatmalik** [comment]
Discussed this in refinement 3/Oct - agreed to do i.e. preview any candidate attachment in TC without download.

**Comment by cazcam34** [comment]
There seems to be a simple fix - just pointing the link to the google url which opens the link in the browser. However the note (see screenshot) explains that there was a problem doing this, which is why there is a download method... looking into the code history it seems this was done in 2020 by @camerojo. I wonder if maybe our google drive settings have changed since then and now allow us to do the direct link?

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNVFWQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--197ddbf2268ba083c7176933da49679e0e07f9c2/Screen%20Shot%202023-10-09%20at%2011.25.40%20am.png)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNVlWQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d575effee800a5de427c3a4397c3f167ee97fd49/Screen%20Shot%202023-10-09%20at%2011.28.37%20am.png)

**Comment by camerojo** [comment]
@cazcam34 - Interesting - I can't recall the exact problem I was having back then with Google Drive.

Of course if you are using the TC while logged on to Google with your TBB email, you will be able to follow the Google link. Have you tested the link when you are not logged in to Google?

**Comment by cazcam34** [comment]
Have spent a long time trying to figure out how to go to the google drive link WITH the authentication. Going through the files.get(id) method in the GoogleFileSystemServiceIml with different methods (execute(), executeMedia(), fields("webContentLink"), buildHttpRequest(), buildHttpRequestUrl() - pretty much all of them!!). 

I can't figure out how to display the google file in a new browser OR turn the google http request into a spring http request (eg. ResponseEntity). The google HttpRequest & HttpHeaders are different to the spring objects and I have tried converting with not alot of success.

Everytime I get bogged down I realise it should be simple.... I have the drive link I just need to add the authentication headers!! 

Any help much appreciated. 

Some of the resources I have used:
https://developers.google.com/drive/api/reference/rest/v3/files#File
https://developers.google.com/drive/api/reference/rest/v3/files/get
https://developers.google.com/drive/api/guides/manage-downloads
